### ISSUE TITLE
Lens reopens port-forwarding sessions, which presumably leads to "No buffer space available" 

### ISSUE THREAD

**Issue by antonmatsiuk** [kontena/lens#6601]
**Describe the bug**
"Eventual "No buffer space available" error and network disconnects on MacOS M1 Chips when Lens has several port-forwarding sessions open"

**To Reproduce**
1. Open multiple port-forwarding sessions >10 in several clusters e.g. to Alert-Manager, Prometheus etc.
2. Restart Lens, reconnect to the clusters. Lens will reconnect all port-forwarding sessions on a start-up
3. Wait for some time
4. Eventually, internet connection gets broken with "No buffer space available" message in terminal or in a browser.
5. Exit Lens, internet connection is restored.

**Expected behavior**
Lens has an option to not restore port-forwarding sessions on a cluster reconnect.
Port forwarding sessions do not lead to "No buffer space available" error

**Environment (please complete the following information):**
- Lens Version:
- OS: MacOS 13.0.1 on M1 Chip
- Installation method: Mac OS Apple Silicon (.dmg)

**Comment by Nokel81** [comment]
Is it that after restart there are now 20 port forwarding sessions or that you stopped some of them before restarting but after restart Lens tries to start all of them?

**Comment by antonmatsiuk** [comment]
@Nokel81 exactly, port forwarding sessions just stay forever if you do not close them explicitly, and they're re-opened after restart.  f you manage 20+ clusters, you can easily end up with dozens of port-forwarding sessions for which there is neither an overview nor an option to kill them after shutdown.

**Comment by jim-docker** [comment]
I can't reproduce, though I have Intel not M1. When you disconnect from a cluster all port-forward sessions on the cluster should be stopped and the local ports freed up.

I don't quite understand though. If a port-forwarding session is not stopped after quitting Lens, and then you restart Lens and connect to the cluster, then when Lens tries to restart the port-forward it should fail (because it will try to use the same port, which would already be in use if the previous port-forwarding session was not stopped). Something else may be going on.

**Comment by antonmatsiuk** [comment]
@jim-docker I double-checked and yes, you're right, I've modified the issue. The problem is that Lens reconnects all port forwarding sessions once you connect to the cluster again, and this sporadically leads to "No buffer space available" error after some time. Cleaning all port forwarding sessions manually on disconnect is cumbersome if you operate 10+ clusters. One option is to have a setting in Lens to skip automatic port forwarding reconnect on cluster connection. Another is to investigate the buffer overflow issue, which might be caused by kubectl not properly releasing network buffers.

**Comment by jim-docker** [comment]
Thanks for the clarification. What do you mean by "Cleaning all port forwarding sessions manually on disconnect". Do you mean disabling them in the Lens UI? Or do you see some evidence of the `kubectl port-forward` processes from outside of Lens?

**Comment by antonmatsiuk** [comment]
@jim-docker no there are no `kubectl port-forward` outside of Lens, the problem if, I meant that you need to close all port-forwarding sessions in the UI in every cluster manually. You do not remember about them next time you start Lens and connect to the cluster(s), but the port forwarding sessions start in the background. This is not desired due to the reason described above.

**Comment by agarkoff** [comment]
Сталкиваюсь с такой же проблемой.
```
java.net.SocketException: No buffer space available (maximum connections reached?): connect
	at java.base/sun.nio.ch.Net.connect0(Native Method)
	at java.base/sun.nio.ch.Net.connect(Net.java:483)
	at java.base/sun.nio.ch.Net.connect(Net.java:472)```

Lens 2025.2.141554-latest
